### PR TITLE
Review Outcomes

### PR DESCRIPTION
**Pull request by iw-gtadam**
- Addding Cognito service integration to diagram.
- Updating EFS security group tag name.
- Removing unnecessary EFS VPC Endpoint.
- Tightening loadbalancer security group rules.
- Splitting IAM roles to each service and tightening to only what is required for each service.
- Encrypt monitoring bucket.
- Run grafana container as non-root user.

More details in ticket: DW-4530

### CODE REVIEW

File: ecs_grafana.tf [hunk 1]
```diff
@@ -148,3 +148,69 @@ resource "aws_security_group_rule" "allow_loadbalancer_ingress_grafana_http" {
...
+
+resource "aws_iam_role" "grafana" {
+  count              = local.is_management_env ? 1 : 0
+  name               = "grafana"
+  assume_role_policy = data.aws_iam_policy_document.grafana[local.primary_role_index].json
+  tags               = merge(local.tags, { Name = "grafana" })
+}
+
+data "aws_iam_policy_document" "grafana" {
+  count = local.is_management_env ? 1 : 0
```
Comment (mattburgess) [inline ecs_grafana.tf:160]: You don't need a count here as `data` sources aren't real resources. This then simplifies the reference to it on line 155 above. This goes for all other data sources that follow too.

File: ecs_grafana.tf [hunk 2]
```diff
@@ -148,3 +148,69 @@ resource "aws_security_group_rule" "allow_loadbalancer_ingress_grafana_http" {
...
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy" "grafana" {
```
Comment (mattburgess) [inline ecs_grafana.tf:173]: We've got a general pattern used elsewhere of making roles and policies separate resources, then tying them together with `aws_iam_role_policy_attachment`s. That way, the same policy can be referenced by another role, if required.

File: ecs_grafana.tf [hunk 3]
```diff
@@ -148,3 +148,69 @@ resource "aws_security_group_rule" "allow_loadbalancer_ingress_grafana_http" {
...
 }
+
+resource "aws_iam_role" "grafana" {
+  count              = local.is_management_env ? 1 : 0
+  name               = "grafana"
+  assume_role_policy = data.aws_iam_policy_document.grafana[local.primary_role_index].json
+  tags               = merge(local.tags, { Name = "grafana" })
+}
+
+data "aws_iam_policy_document" "grafana" {
```
Comment (mattburgess) [inline ecs_grafana.tf:159]: General naming convention for these types of policy docs is `grafana_assume_role` to differentiate it from other policies attached to the role.

File: ecs_prometheus.tf [hunk 4]
```diff
@@ -181,11 +181,152 @@ resource "aws_security_group_rule" "allow_prometheus_egress_https" {
...
   source_security_group_id = aws_security_group.efs.id
 }
+
+resource "aws_iam_role" "prometheus" {
+  name               = "prometheus"
+  assume_role_policy = data.aws_iam_policy_document.prometheus.json
+  tags               = merge(local.tags, { Name = "prometheus" })
+}
+
+data "aws_iam_policy_document" "prometheus" {
```
Comment (mattburgess) [inline ecs_prometheus.tf:199]:
```suggestion
data "aws_iam_policy_document" "prometheus_assume_role" {
```

File: ecs_thanos.tf [hunk 5]
```diff
@@ -144,3 +144,69 @@ resource "aws_security_group_rule" "allow_grafana_ingress_thanos_http" {
...
 }
+
+resource "aws_iam_role" "thanos" {
+  count              = local.is_management_env ? 1 : 0
+  name               = "thanos"
+  assume_role_policy = data.aws_iam_policy_document.thanos[local.primary_role_index].json
+  tags               = merge(local.tags, { Name = "thanos" })
+}
+
+data "aws_iam_policy_document" "thanos" {
```
Comment (mattburgess) [inline ecs_thanos.tf:155]:
```suggestion
data "aws_iam_policy_document" "thanos_assume_role" {
```

File: ecs_grafana.tf (hunk 1 above)
Comment (iw-gtadam) [reply ecs_grafana.tf]: Resolved

File: ecs_grafana.tf (hunk 2 above)
Comment (iw-gtadam) [reply ecs_grafana.tf]: Resolved

File: ecs_grafana.tf (hunk 3 above)
Comment (iw-gtadam) [reply ecs_grafana.tf]: Resolved

File: ecs_prometheus.tf (hunk 4 above)
Comment (iw-gtadam) [reply ecs_prometheus.tf]: Resolved

File: ecs_thanos.tf (hunk 5 above)
Comment (iw-gtadam) [reply ecs_thanos.tf]: Resolved

**Pull request merged**